x-pack/plugins/upgrade_assistant/public/application/components/overview/fix_logs_step/external_links.tsx

```diff
@@ -33,10 +33,18 @@ const getDeprecationIndexPatternId = async (dataService: DataPublicPluginStart)
   if (deprecationIndexPattern) {
     return deprecationIndexPattern.id;
   } else {
+    // When creating the index pattern, we need to be careful when creating an indexPattern
+    // for an index that doesnt exist. Since the deprecation logs data stream is only created
+    // when a deprecation log is indexed it could be possible that it might not exist at the
+    // time we need to render the DiscoveryAppLink.
+    // So in order to avoid those errors we need to make sure that the indexPattern is created
+    // with allowNoIndex and that we skip fetching fields to from the source index.
+    // prettier-ignore
     const newIndexPattern = await dataService.dataViews.createAndSave({
       title: DEPRECATION_LOGS_INDEX_PATTERN,
       allowNoIndex: true,
-    });
+    }, false, true);
```

Comment:
To improve readability, WDYT of creating variables for these?

```ts
const override = false;
const skipFetchFields = true;

const newIndexPattern = await dataService.dataViews.createAndSave({
  title: DEPRECATION_LOGS_INDEX_PATTERN,
  allowNoIndex: true,
}, override, skipFetchFields);
```

Reply:
good one! I've just pushed a lil commit for that